### ISSUE TITLE
sp_before_unnamed_ptr_star=remove removes space before named ptr_star

### ISSUE THREAD

**Issue by rousskov** [uncrustify/uncrustify#402]
When I add `sp_before_unnamed_ptr_star=remove` to the configuration file,

```
Adapter::Answer *Adapter::Answers::get()
{
}
```

is formatted as

```
Adapter::Answer*Adapter::Answers::get()
{
}
```

instead of remaining the same.

FWIW, the combination of ptr_star options I would like to use is:

```
sp_before_ptr_star=add
sp_before_ptr_star_func=add
sp_before_unnamed_ptr_star=remove
sp_between_ptr_star=remove
```

but I cannot have `sp_before_unnamed_ptr_star=remove` there because of this bug.

Tested with commit fc5228e.

**Comment by rousskov** [comment]
The problem is still there as of commit 321fc5a (v0.62+). More examples of unwanted side-effects of `sp_before_unnamed_ptr_star=remove` (in diff format):

```
-void *operator new(size_t size) throw (std::bad_alloc)
+void*operator new(size_t size) throw (std::bad_alloc)
```

```
-        ProxyCfg *const proxyCfg = TheProxySharedCfgs.getConfig(proxy);
+        ProxyCfg*const proxyCfg = TheProxySharedCfgs.getConfig(proxy);
```

Please let me know if you need more information about this problem.

**Comment by gmaurel** [comment]
This is correct.

The description says:
"Add or remove space before pointer star '*' that isn't followed by a variable name
 If set to 'ignore', sp_before_ptr_star is used instead."

The option sp_before_unnamed_ptr_star works before sp_before_ptr_star.

**Comment by rousskov** [comment]
It sounds like your interpret "is not followed by a variable name" as "is not followed by a variable name, including contexts where no variable name can be present at all and contexts where variable name does follow but not immediately". That [arguably valid] interpretation seems to make sp_before_unnamed_ptr_star impossible to use without undesirable side effects. Please let me know if there is another option that removes spaces before pointer stars in unnamed function parameters without these side effects.

In other words, is there a way to get

```diff
-void foo(pointer_t *);
+void foo(pointer_t*);
```

without also getting

```diff
-void *operator new(size_t size) throw (std::bad_alloc)
+void*operator new(size_t size) throw (std::bad_alloc)
```

and

```diff
-        ProxyCfg *const proxyCfg = TheProxySharedCfgs.getConfig(proxy);
+        ProxyCfg*const proxyCfg = TheProxySharedCfgs.getConfig(proxy);
```
